src/lib/core/say.h

```diff
@@ -483,6 +483,19 @@ say_format_plain(struct log *log, char *buf, int len, int level,
 		 const char *filename, int line, const char *error,
 		 const char *format, va_list ap);
 
+/**
+ * A type defining a callback that is called before or after
+ * writing to stderr.
+ */
+typedef void
+(*say_stderr_write_cb_t)(void);
+
+/**
+ * Set callback functions called before/after writing to stderr.
+ */
+void
+say_stderr_write_cb(say_stderr_write_cb_t before, say_stderr_write_cb_t after);
```

Comment:
I think, It would be nice to have `set` in the name of a setter function, because I expect function with this name get `cb` and not set it.

Reply:
Renamed to `say_set_stderr_callback()`. Renamed `say_stderr_write_cb_t` to `say_stderr_callback_t` for consistency.

<details>
<summary>Diff</summary>

```diff
diff --git a/src/lib/core/say.c b/src/lib/core/say.c
index 6c96363cd..1602f0f6f 100644
--- a/src/lib/core/say.c
+++ b/src/lib/core/say.c
@@ -91,8 +91,8 @@ say_format_syslog(struct log *log, char *buf, int len, int level,
 /*
  * Callbacks called before/after writing to stderr.
  */
-static say_stderr_write_cb_t before_stderr_write_cb;
-static say_stderr_write_cb_t after_stderr_write_cb;
+static say_stderr_callback_t before_stderr_callback;
+static say_stderr_callback_t after_stderr_callback;
 
 /** Default logger used before logging subsystem is initialized. */
 static struct log log_boot = {
@@ -1282,11 +1282,11 @@ log_vsay(struct log *log, int level, const char *filename, int line,
 		write_to_file(log, total);
 		break;
 	case SAY_LOGGER_STDERR:
-		if (before_stderr_write_cb != NULL)
-			before_stderr_write_cb();
+		if (before_stderr_callback != NULL)
+			before_stderr_callback();
 		write_to_file(log, total);
-		if (after_stderr_write_cb != NULL)
-			after_stderr_write_cb();
+		if (after_stderr_callback != NULL)
+			after_stderr_callback();
 		break;
 	case SAY_LOGGER_SYSLOG:
 		write_to_syslog(log, total);
@@ -1295,12 +1295,12 @@ log_vsay(struct log *log, int level, const char *filename, int line,
 		break;
 	case SAY_LOGGER_BOOT:
 	{
-		if (before_stderr_write_cb != NULL)
-			before_stderr_write_cb();
+		if (before_stderr_callback != NULL)
+			before_stderr_callback();
 		ssize_t r = safe_write(STDERR_FILENO, buf, total);
 		(void) r;                       /* silence gcc warning */
-		if (after_stderr_write_cb != NULL)
-			after_stderr_write_cb();
+		if (after_stderr_callback != NULL)
+			after_stderr_callback();
 		break;
 	}
 	default:
@@ -1311,8 +1311,9 @@ log_vsay(struct log *log, int level, const char *filename, int line,
 }
 
 void
-say_stderr_write_cb(say_stderr_write_cb_t before, say_stderr_write_cb_t after)
+say_set_stderr_callback(say_stderr_callback_t before,
+			say_stderr_callback_t after)
 {
-	before_stderr_write_cb = before;
-	after_stderr_write_cb = after;
+	before_stderr_callback = before;
+	after_stderr_callback = after;
 }
diff --git a/src/lib/core/say.h b/src/lib/core/say.h
index f40e26e82..6c9af96b9 100644
--- a/src/lib/core/say.h
+++ b/src/lib/core/say.h
@@ -488,13 +488,14 @@ say_format_plain(struct log *log, char *buf, int len, int level,
  * writing to stderr.
  */
 typedef void
-(*say_stderr_write_cb_t)(void);
+(*say_stderr_callback_t)(void);
 
 /**
  * Set callback functions called before/after writing to stderr.
  */
 void
-say_stderr_write_cb(say_stderr_write_cb_t before, say_stderr_write_cb_t after);
+say_set_stderr_callback(say_stderr_callback_t before,
+			say_stderr_callback_t after);
 
 #if defined(__cplusplus)
 } /* extern "C" */
```
</details>